Add STASH Unit build requirements to clue scroll overlay

DavidLekei opened the pull request:
Suggested by #14744

CFitzsimons commented:
My 2c would be that this should probably be a menu option and not enabled by default.  Thoughts?

DavidLekei commented:
Added a menu option to enable displaying the build requirements (default is disabled). Also fixed multiple of any type of item requirement being counted correctly (ie building requires 10 of any type of nails, but quantity wasn't being checked properly)
